### PR TITLE
Update file

### PR DESCRIPTION
**Pull request by mihaisc**
to test this : 
- use rinkeby
- publish a file in market (not this branch) or just take one older published asset (version 4.0.0)
- edit asset and change file url
- check if asset is downloadable and that version is 4.1.0

### CODE REVIEW

**Comment by kremalicious** [comment]
this asset worked on 2nd try again after failing at final download with that known provider issue
https://market-git-fix-updatefilestructure-oceanprotocol.vercel.app/asset/did:op:4a89a8cbd80d08080924ed85b3751c16b63085166c78e2f2a9a3a8ea0944263a

Tried publish new one, worked well, just slow to index in end, was only showing up after like 1-2 min.
https://market-git-fix-updatefilestructure-oceanprotocol.vercel.app/asset/did:op:ff882dc0147a180d2d6ca1a9bc61fe6fa9779af2517924afafc1d7e3a3f1e53b

Download shows same `sender of order transaction is not the consumer/payer` provider issue. I have to do full browser page refresh to get Download to work in end. But as far as this PR goes, seems to be all alright

**Comment by kremalicious** [comment]
that edit form is a UX nightmare but guess we keep it for #1244 where I added that issue which is made worse through adding the file input. Makes it hard to know what field gets edited or cleared. So both sample & file should start out in "added" mode which has to be dismissed to change. Current implementation will make people think the file & sample gets cleared based on the other form elements. Quick fix is to be consistent with the labels and also use "New ..." on file & sample

**Comment by bogdanfazakas** [comment]
> Works flawlessly on Rinkeby, but got some issues on Ropsten & Polygon: the file url it's not updated, on download I still get the old file

I think the provider version needed for this was just deployed Rinkeby, also but for aquarius it takes some time to index the updated metadata so you probably can encounter this if the update was not indexed when the asset was reloaded

**Comment by kremalicious** [comment]
Should only work on Rinkeby right now, as only there new provider is deployed so seems all good. Also tested again on Rinkeby, verified updated file is actually updated and can be downloaded. All flawless

**Comment by EnzoVezzaro** [comment]
I also confirm it's all good on Rinkeby, just tested.

**Review by EnzoVezzaro — APPROVED** [review]

**Review by claudiaHash — APPROVED** [review]

**Comment by mihaisc** [comment]
Not too fond about deleting props but... it works 😁

File: src/components/Asset/Edit/EditMetadata.tsx [hunk 1]
```diff
@@ -78,17 +80,43 @@ export default function Edit({
...
+      }
       const updatedService: Service = {
         ...asset.services[0],
-        timeout: mapTimeoutStringToSeconds(values.timeout)
+        timeout: mapTimeoutStringToSeconds(values.timeout),
+        files: updatedFiles
       }
 
+      // TODO: remove version update at a later time
       const updatedAsset: Asset = {
```
Comment (kremalicious) [inline src/components/Asset/Edit/EditMetadata.tsx:109]: on first look seems we could have spotted this issue earlier if we would use the `DDO` typing here instead?

```ts
const updatedAsset: DDO
```

Comment (mihaisc) [reply src/components/Asset/Edit/EditMetadata.tsx:109]: nope, typescript does not remove the props. `Asset`  doesn't have accessDetails and yet it's still there.

**Pull request merged**